pandas/tests/groupby/methods/test_size.py

```diff
@@ -76,18 +74,16 @@ def test_size_series_masked_type_returns_Int64(dtype):
     tm.assert_series_equal(result, expected)
 
 
-# TODO(infer_string) in case the column is object dtype, it should preserve that dtype
-# for the result's index
-@pytest.mark.xfail(using_string_dtype(), reason="TODO(infer_string)", strict=False)
-def test_size_strings(any_string_dtype):
+def test_size_strings(any_string_dtype, using_infer_string):
     # GH#55627
     dtype = any_string_dtype
     df = DataFrame({"a": ["a", "a", "b"], "b": "a"}, dtype=dtype)
     result = df.groupby("a")["b"].size()
     exp_dtype = "Int64" if dtype == "string[pyarrow]" else "int64"
+    exp_index_dtype = "str" if using_infer_string and dtype == "object" else dtype
     expected = Series(
         [2, 1],
-        index=Index(["a", "b"], name="a", dtype=dtype),
+        index=Index(["a", "b"], name="a", dtype=exp_index_dtype),
```

Comment:
Why doesn't it work if you just remove the `dtype` argument and let the constructor infer?

Reply:
Good question - this was introduced in https://github.com/pandas-dev/pandas/pull/55627 but I do not see why if the values are `string[pyarrow]` that the result would be `Int64`.

cc @phofl

Reply:
@rhshadrach the Int64 is for `exp_dtype` on the line below, not for the dtype of the Index being constructed on this line, so I am not entirely understanding your comment/question ? 
(the construction of `exp_dtype` is not being touched in this PR)

Reply:
Ah, thanks! When the grouping column is StringDtype, we preserve this even when infer_string is False. In the groupby code, the uniques that go into creating the index is a string array. When the input is object dtype and `infer_string=True`, we do inference on the fails and coerce to dtype `str`.

So in the object case we're doing inference, whereas in the non-object case we are not. It seems reasonable to me, thoughts?

Reply:
Ah OK thanks for the explanation. I am not sure how I feel yet, but at first I wasn't expecting the action of grouping to perform any inference. Is that not a performance hit?

Reply:
FWIW, if this is not easy to "fix" (avoid the inference), then I am personally fine with the current behaviour for now

Reply:
Great - I think we are all leaning in that direction

Reply:
The line in question is

https://github.com/pandas-dev/pandas/blob/e3b2de852a87dc7b530302e0039730e7745b2fcf/pandas/core/groupby/ops.py#L755

While it's been moved around recently, I believe that's long standing behavior. I can investigate what impact removing that (so just using standard `Index` init) would have, but seems independent.